core/camel-support/src/main/java/org/apache/camel/support/PropertyBindingSupport.java

```diff
@@ -287,9 +323,22 @@ public static boolean bindProperties(CamelContext camelContext, Object target, M
 
         for (Iterator<Map.Entry<String, Object>> iter = properties.entrySet().iterator(); iter.hasNext();) {
             Map.Entry<String, Object> entry = iter.next();
-            if (bindProperty(camelContext, target, entry.getKey(), entry.getValue(), nesting, deepNesting, fluentBuilder, reference, placeholder)) {
-                iter.remove();
-                rc = true;
+            String key = entry.getKey();
+            Object value = entry.getValue();
+
+            if (isNotEmpty(optionPrefix)) {
+                if (!key.startsWith(optionPrefix)) {
+                    return false;
```

Comment:
I think this is wrong, we should just skip keys without the prefix, and not return

Reply:
oh yes, taken fomr the introspection support, thx for spotting